### PR TITLE
Frontend: playlist modification actions

### LINKED ISSUE #21: Frontend: playlist creation/modification actions
[`frontend--playlist-mod-actions`](https://github.com/masonmcelvain/geomusic/compare/frontend--playlist-mod-actions)

We want to let users do 2 things:
* Create a playlist
* Add to a playlist

#### Create
Create a dummy link from the landing page that opens a 'Create playlist' view. The view should let the user specify a name for the new playlist. After committing the name, the view will transition to an empty 'Add to playlist' page.

#### Add
Modify the view playlist page to allow for adding songs. This will involve adding a 'add' button to the screen, which will take the user to a search view. The user will type in the song name/ artist name and be able to select a song from the list. For now, the user will only be allowed to add a single song at a time.

Estimate: 3 hours

### PR DESCRIPTION
**Pull request by masonmcelvain**
Added some UI features to create a new playlist and add songs to an existing playlist.

The create playlist button opens a modal that allows the user to specify the title for a new playlist. The user is then navigated to the view playlist page.

The view playlist page gets a new '+' icon in the upper right that lets the user add songs to the playlist through a popup search modal. Currently, the search utility doesn't work because we need to set up backend routes to implement that functionality.

Closes #21

### CODE REVIEW

**Comment by scole02** [comment]
Great job with the header, I would move the search bar up to the top to reduce the negative space.

**Review by scole02 — COMMENTED** [review]
Great job with the header, I would move the search bar up to the top to reduce the negative space.

**Comment by masonmcelvain** [comment]
Addressed Sam's comment with https://github.com/masonmcelvain/geomusic/pull/23/commits/23856df69bedc3b2d79f6d9c0aa411a2b3f2c2bc

File: navigation/index.tsx [hunk 1]
```diff
@@ -4,13 +4,18 @@ import {
   DarkTheme,
 } from "@react-navigation/native";
 import { createStackNavigator } from "@react-navigation/stack";
-import * as React from "react";
+import React, { useState } from "react";
 import { ColorSchemeName } from "react-native";
 
 import BrowseScreen from "../screens/BrowseScreen";
+import CreatePlaylistModal from "../screens/CreatePlaylistModal";
```
Comment (kylene-phillips) [inline navigation/index.tsx:11]: Modals are a good addition differentiating actions for pop-up windows and screens. Great modification.

File: navigation/index.tsx [hunk 2]
```diff
@@ -28,27 +33,80 @@ export default function Navigation({
...
           title: route.params.title,
           playlistId: route.params.playlistId,
         })}
       />
-      <Stack.Screen
+    </MainStack.Navigator>
+  );
+}
+
+function RootNavigator() {
```
Comment (kylene-phillips) [inline navigation/index.tsx:62]: Root Nav is helpful for keeping all the screens and modals organized and see what is returned. Clean

File: json/sample_songs.json [hunk 3]
```diff
@@ -0,0 +1,92 @@
+[
```
Comment (kylene-phillips) [inline json/sample_songs.json:1]: Looking forward to linking this with the backend + db this weekend

File: components/SongScrollColumn.tsx [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+}
+
+type SongRowProps = {
+  song: Song;
+  handleSongPress: () => void;
+};
+
+function SongRow({ song, handleSongPress }: SongRowProps) {
+  return (
+    <TouchableHighlight
```
Comment (kylene-phillips) [inline components/SongScrollColumn.tsx:41]: I like touchable highlight's functionality and options much more than what we had before

**Review by kylene-phillips — COMMENTED** [review]
See comments on several of the individuals commits. I like the organization, Root Nav, and addition of button configs. Looking forward to finishing the routes and login page to put this all together. Great work mason.

**Pull request merged**